pkg/diff/precedence.go

```diff
@@ -104,6 +97,41 @@ func ValidateManager(reconciler, manager string, id core.ID, op admissionv1.Oper
 		klog.Warningf("Invalid manager annotation (object: %q, annotation: %s=%q)", id, metadata.ResourceManagerKey, manager)
 	}
 
+	switch id.GroupKind {
+	case kinds.RootSyncV1Beta1().GroupKind():
+		if isReconcilerManager(reconciler) {
+			if op == admissionv1.Update {
+				// ReconcilerManager is allowed to update RootSyncs to add/remove the finalizer
+				return nil
+			}
+		}
+		if core.RootReconcilerName(id.Name) == reconciler {
+			if op == admissionv1.Update {
+				// root-reconciler is allowed to update its own RootSync to add/remove the finalizer
+				return nil
+			}
+			// root-reconciler is NOT allowed to create or delete its own RootSync
+			return fmt.Errorf("config sync %q can not %s object %q managed by config sync %q",
+				reconciler, op, id, oldReconciler)
+		}
+	case kinds.RepoSyncV1Beta1().GroupKind():
+		if isReconcilerManager(reconciler) {
+			if op == admissionv1.Update {
+				// ReconcilerManager is allowed to update RootSyncs to add/remove the finalizer
+				return nil
+			}
+		}
+		if core.NsReconcilerName(id.Namespace, id.Name) == reconciler {
+			if op == admissionv1.Update {
+				// ns-reconciler is allowed to update its own RepoSync to add/remove the finalizer
+				return nil
+			}
+			// root-reconciler is NOT allowed to create or delete its own RootSync
```

Comment:
comment should be `// ns-reconciler is NOT allowed to create or delete its own RepoSync`